gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java

```diff
@@ -337,4 +376,45 @@ public void When_MedicationRequestMapperCompletesSuccessfully_Expect_MedicationM
     private RCMRMT030101UKEhrExtract unmarshallEhrExtract(String fileName) {
         return unmarshallFile(getFile("classpath:" + XML_RESOURCES_BASE + fileName), RCMRMT030101UKEhrExtract.class);
     }
-}
+
+    private void assertMedicationRequests(List<? extends DomainResource> resources) {
+        resources.stream()
+            .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+            .map(MedicationRequest.class::cast)
+            .forEach(this::assertMedicationRequestIdIsPatientId);
+    }
+
+    private void assertMedicationStatements(List<? extends DomainResource> resources) {
+        resources.stream()
+            .filter(resource -> ResourceType.MedicationStatement.equals(resource.getResourceType()))
+            .map(MedicationStatement.class::cast)
+            .forEach(this::assertMedicationStatementDateAndIdIsPatient);
+    }
+
+    private void assertMedicationStatementDateAndIdIsPatient(MedicationStatement statement) {
+        final Date date = statement.getDateAssertedElement().getValue();
+
+        assertMedicationStatementIdIsPatientId(statement);
+        assertThat(date).isEqualTo(EXPECTED_DATE_TIME_TYPE.getValue());
+    }
+
+    private void assertMedicationStatementIdIsPatientId(MedicationStatement medicationStatement) {
```

Comment:
I think here you can do something like this in order to eliminate few lines of code:
private void assertMedicationStatementIdIsPatientId(MedicationStatement medicationStatement) {        
        assertResourceIdEqualsPatientId(medicationStatement.getSubject());
    }

    private void assertMedicationRequestIdIsPatientId(MedicationRequest medicationRequest) {     
        assertResourceIdEqualsPatientId(medicationRequest.getSubject());
    }

    private void assertResourceIdEqualsPatientId(Reference ref) {
        final String resourceId = ref            
            .getResource()
            .getIdElement()
            .getIdPart();

        assertEquals(PATIENT_ID, resourceId);
    }

but it is just an option...